test/image/mocks/bar_attrs_group_norm.json

```diff
@@ -1,19 +1,24 @@
 {
     "data":[
         {
-            "base":4,
-            "x":[3,2,1,0],
+            "text":0.5,
+            "textposition":"outside",
+            "outsidetextfont": {"size":32},
+            "base":[100,40,25,10],
+            "x":[-50,10,25,40],
             "type":"bar"
         }, {
-            "base":[7,6,5,4],
-            "x":[1,2,3,4],
+            "text":0.5,
```

Comment:
That could be an option - for conformity with tick labels it should be `textangle` - though offhand I'd say it opens more questions than it answers. For example, you might want 45 degrees or something for text on vertical bars in order to make the labels more readable, but have the labels expand into the space of the next bar(s) - similar to how angles are used on tick labels. But then because these labels start from different vertical points, making it so they don't overlap becomes challenging.  I would leave this out for now, and then when we do add it, this will override the "auto" rotation that we determine now.

So then thinking a little more about what this auto behavior should be - basically I feel like text should be horizontal unless it can be less shrunken being vertical - similar to the inside/outside criterion. I think that's already what you're doing for vertical bars, but for horizontal it looks like you're defaulting to vertical text when I think 99% of the time it should be horizontal, and then in those cases that you *do* decide to rotate the text, it should only turn 90 degrees clockwise.